CONTRIBUTING.md

```diff
@@ -616,8 +616,26 @@ documentation][coqdoc-documentation] to learn more.
 
 ### Fixing bugs and performing small changes ###
 
-Just open a PR with your fix.  If it is not yet completed, do not
-hesitate to open a [*draft PR*][GitHub-draft-PR] to get early
+Before fixing a bug, it is best to check that it was reported before:
+
+- If it was already reported and you intend to fix it, self-assign the
+  issue (if you have the permission), or leave a comment marking your
+  intention to work on it (and a contributor with write-access may
+  then assign the issue to you).
+
+- If the issue already has an assignee, you should check with them if
+  they still intend to work on it.  If the assignment is several
+  weeks, months, or even years (!) old, there are good chances that it
+  does not reflect their current priorities.
+
+- If the bug has not been reported before, it can be a good idea to
+  open an issue about it, while stating that you are preparing a fix.
+  The issue can be the place to discuss about the bug itself while the
+  PR will be the place to discuss your proposed fix.
+
```

Comment:
Maybe a good subjective "hint" here about whether to open an issue about a problem you have detected but is unnoticed so far is how likely are other to encounter the bug in the future. IMHO if that's likely, then an issue is good so people can search, otherwise, we can just recommend documenting the problem in the PR.

Reply:
I'll merge, however I still think we could make this tweak.

Reply:
The reasoning was more: advertise what you are working on, especially if you are an external contributor, as you could get valuable information that you would have missed otherwise.